frontend/app/routes/$lang+/_protected+/letters+/index.tsx

```diff
@@ -100,6 +100,7 @@ export default function LettersIndex() {
       </div>
       <ul className="divide-y border-y">
         {letters.map((letter) => {
+          if (letter.name === undefined || letter.name === '') return;
```

Comment:
A cleaner way to do this would be to use `.filter(..)`:

``` tsx
letters
  .filter(letter => letter.name?.length > 0)
  .map(..)
```

But if all of your letters have undefined or empty names, you will be left with an empty `<ul></ul>`... so my suggestion would be to filter inside of the component code (ie: before the `return`) and then conditionally render either the `<ul>` or `<p>There are no letters on file.</p>` (or something similar).